index.js

```diff
@@ -87,22 +94,23 @@ module.exports = function(opts) {
 				}
 
 				// listening port is chosen by NodeJS if opts.port === 0
-				const port = server.address().port;
-
-				broker.healthcheck.port = port;
-				broker.healthcheck.emit('port', port);
+				broker.healthcheck.port = server.address().port;
+				broker.healthcheck.emit('port', broker.healthcheck.port);
 
 				broker.logger.info('');
 				broker.logger.info('K8s health-check server listening on');
-				broker.logger.info(`    http://localhost:${port}${opts.readiness.path}`);
-				broker.logger.info(`    http://localhost:${port}${opts.liveness.path}`);
+				broker.logger.info(`    http://localhost:${broker.healthcheck.port}${opts.readiness.path}`);
```

Comment:
This modification does not improve readability, it rather make it harder to read the code.

Reply:
Agreed, this was not an intended change.

Reply:
Fixed in a761d62